server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -187,6 +187,11 @@ public KeywordFieldMapper build(BuilderContext context) {
         }
     }
 
+    @Override
+    protected Object parseSourceValue(Object value) {
```

Comment:
nit: I was wondering if we should use covariance here and anywhere we return Strings, like we do for other types.

Reply:
👍